modules/data.land/R/download.SM_CDS.R

```diff
@@ -0,0 +1,91 @@
+#' Download CDS soil moisture data for the SDA workflow.
+#'
+#' @param outfolder physical paths to where the unziped soil moisture files are downloaded.
+#' @param time_points A vector contains each time point within the start and end date.
+#' @param overwrite flag determine if we want to overwrite existing files when downloading.
+#'
+#' @return A vector containing file paths to the downloaded files.
+#' @export
+#' 
+#' @examples
+#' @author Dongchen Zhang
+#' @importFrom magrittr %>%
+download.SM_CDS <- function(outfolder, time.points, overwrite = FALSE) {
+  #load cdsapi from python environment.
+  tryCatch({
+    cdsapi <- reticulate::import("cdsapi")
+  }, error = function(e) {
+    PEcAn.logger::logger.severe(
+      "Failed to load `cdsapi` Python library. ",
+      "Please make sure it is installed to a location accessible to `reticulate`.",
+      "You should be able to install it with the following command: ",
+      "`pip install --user cdsapi`.",
+      "The following error was thrown by `reticulate::import(\"cdsapi\")`: ",
```

Comment:
This sort of dependency on a python library needs to documented in your Roxygen metadata. 

I also found this article on managing R library dependencies on python libraries: https://cran.r-project.org/web/packages/reticulate/vignettes/python_dependencies.html